webportal/src/app/user/reset-password/reset-password.component.js

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+// module dependencies
+const breadcrumbComponent = require('../../job/breadcrumb/breadcrumb.component.ejs');
+const resetPasswordComponent = require('./reset-password.component.ejs');
+const webportalConfig = require('../../config/webportal.config.json');
+const userAuth = require('../user-auth/user-auth.component');
+require('./reset-password.component.scss');
+
+
+const resetPasswordHtml = resetPasswordComponent({
+  breadcrumb: breadcrumbComponent
+});
+
+$('#content-wrapper').html(resetPasswordHtml);
+$(document).ready(() => {
+  $('#form-reset-password').on('submit', (e) => {
+    e.preventDefault();
+    const username = $('#form-reset-password :input[name=username]').val();
```

Comment:
Why do we have to input the user name?

Reply:
And I suggest we let users type in: 1) the old password, 2) the new password, and 3) the new password again, as Github (and most of other systems) does:
![image](https://user-images.githubusercontent.com/7257612/35802611-bcad8c00-0aab-11e8-91d7-dfd2bfcd5564.png)

Reply:
maybe it is because admin user can change other user's password?

---
In reply to: [165946869](https://github.com/Microsoft/pai/pull/303#discussion_r165946869) [](ancestors = 165946869)

Reply:
This should also apply to the UI that creates a new user.

---
In reply to: [165947446](https://github.com/Microsoft/pai/pull/303#discussion_r165947446) [](ancestors = 165947446)